### PR TITLE
support produce kv-list batch as message set

### PR DESCRIPTION
**Pull request by zmstone**
For messages that are already batched on caller side, there is no need to break down and re-assemble.
Also easier to achieve better compression ratio if compression is enable.

### CODE REVIEW

File: src/brod.erl [hunk 1]
```diff
@@ -248,22 +251,30 @@ get_producer(Client, Topic, Partition) ->
...
-%% @doc Produce one message. The pid should be a producer pid.
--spec produce(pid(), binary(), binary()) ->
+%% @doc Produce one message if Value is binary or iolist,
+%% or a message set if Value is a (nested) kv-list, in this case Key
+%% is discarded (only the keys in kv-list are sent to kafka).
+%% The pid should be a partition producer pid, NOT client pid.
+%% @end
+-spec produce(pid(), key(), value()) ->
         {ok, brod_call_ref()} | {error, any()}.
 produce(ProducerPid, Key, Value) ->
```
Comment (id) [inline src/brod.erl:266]: Perhaps exclude this function from brod: api to not confuse with produce call for brod client?
If a user knows what they are doing, they can just use brod_producer directly.

Comment (id) [inline src/brod.erl:266]: @zmstone let me know if you agree with this and I'll fix it

Comment (zmstone) [reply src/brod.erl:266]: that should be a 3.0 change

Comment (id) [reply src/brod.erl:266]: All right, approved.

**Pull request merged**